uefi/src/proto/shell/mod.rs

```diff
@@ -2,12 +2,208 @@
 
 //! EFI Shell Protocol v2.2
 
-use crate::proto::unsafe_protocol;
+use uefi_macros::unsafe_protocol;
+use uefi_raw::Status;
 
-pub use uefi_raw::protocol::shell::ShellProtocol;
+use core::marker::PhantomData;
+use core::ptr;
+
+use uefi_raw::protocol::shell::ShellProtocol;
+
+use crate::{CStr16, Char16};
 
 /// Shell Protocol
 #[derive(Debug)]
 #[repr(transparent)]
-#[unsafe_protocol(uefi_raw::protocol::shell::ShellProtocol::GUID)]
-pub struct Shell(uefi_raw::protocol::shell::ShellProtocol);
+#[unsafe_protocol(ShellProtocol::GUID)]
+pub struct Shell(ShellProtocol);
+
+/// Iterator over the names of environmental variables obtained from the Shell protocol.
+#[derive(Debug)]
+pub struct Vars<'a> {
+    /// Char16 containing names of environment variables
+    inner: *const Char16,
+    /// Placeholder to attach a lifetime to `Vars`
+    placeholder: PhantomData<&'a CStr16>,
+}
+
+impl<'a> Iterator for Vars<'a> {
+    type Item = &'a CStr16;
+    // We iterate a list of NUL terminated CStr16s.
+    // The list is terminated with a double NUL.
+    fn next(&mut self) -> Option<Self::Item> {
+        let cur_start = self.inner;
+        let mut cur_len = 0;
+        unsafe {
+            if *(cur_start) == Char16::from_u16_unchecked(0) {
+                return None;
+            }
+            while *(cur_start.add(cur_len)) != Char16::from_u16_unchecked(0) {
+                cur_len += 1;
+            }
+            self.inner = self.inner.add(cur_len + 1);
+            Some(CStr16::from_ptr(cur_start))
+        }
+    }
+}
+
+impl Shell {
+    /// Gets the value of the specified environment variable
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The environment variable name of which to retrieve the
+    ///   value.
+    ///
+    /// # Returns
+    ///
+    /// * `Some(<env_value>)` - &CStr16 containing the value of the
+    ///   environment variable
+    /// * `None` - If environment variable does not exist
+    #[must_use]
+    pub fn get_env(&self, name: &CStr16) -> Option<&CStr16> {
```

Comment:
In https://github.com/rust-osdev/uefi-rs/pull/1679#discussion_r2128040189, it was suggested to use the same name as `std`: `var`. I think we should do that here, for several reasons:
* Consistency in APIs is good; anyone who is familiar with `std::env` will instinctively know how to use this API
* Per https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter, Rust APIs generally should not use a `get_` prefix
* It's clearer to say you are getting the variable (noun) rather than the environment (which can be a noun, but in "environment variable" it's an adjective).